android/junit4-examples/src/test/java/com/browserstack/BrowserStackJUnitTest.java

```diff
@@ -1,116 +1,24 @@
-        List<Integer> taskIDs = new ArrayList<Integer>();
-
-        if (System.getProperty("config") != null) {
-            JSONParser parser = new JSONParser();
-            config = (JSONObject) parser.parse(new FileReader("src/test/resources/com/browserstack/" + System.getProperty("config")));
-            int envs = ((JSONArray) config.get("environments")).size();
-
-            for (int i = 0; i < envs; i++) {
-                taskIDs.add(i);
-            }
-        }
-        return taskIDs;
-    }
 
     @Before
     public void setUp() throws Exception {
-        JSONArray envs = (JSONArray) config.get("environments");
-
         UiAutomator2Options options = new UiAutomator2Options();
-
-        Map<String, String> envCapabilities = (Map<String, String>) envs.get(taskID);
-        Iterator it = envCapabilities.entrySet().iterator();
-        while (it.hasNext()) {
-            Map.Entry pair = (Map.Entry) it.next();
-            options.setCapability(pair.getKey().toString(), pair.getValue().toString());
-        }
-
-        Map<String, String> commonCapabilities = (Map<String, String>) config.get("capabilities");
-        it = commonCapabilities.entrySet().iterator();
-        while (it.hasNext()) {
-            Map.Entry pair = (Map.Entry) it.next();
-            if (options.getCapability(pair.getKey().toString()) == null) {
-                options.setCapability(pair.getKey().toString(), pair.getValue());
-            }else if (pair.getKey().toString().equalsIgnoreCase("bstack:options")){
-                HashMap bstackOptionsMap = (HashMap) pair.getValue();
-                bstackOptionsMap.putAll((HashMap) options.getCapability("bstack:options"));
-                options.setCapability(pair.getKey().toString(), bstackOptionsMap);
-            }
-        }
-
-        JSONObject browserstackOptions = (JSONObject) options.getCapability("bstack:options");
-
-        String username = System.getenv("BROWSERSTACK_USERNAME");
-        if (username == null) {
-            username = (String) browserstackOptions.get("userName");
-        }
-
-        String accessKey = System.getenv("BROWSERSTACK_ACCESS_KEY");
-        if (accessKey == null) {
-            accessKey = (String) browserstackOptions.get("accessKey");
-        }
-
-        String app = System.getenv("BROWSERSTACK_APP_ID");
-        if (app != null && !app.isEmpty()) {
-            options.setCapability("app", app);
-        }
-
-        if (browserstackOptions.get("local") != null && browserstackOptions.get("local").toString() == "true") {
-            local = new Local();
-            Map<String, String> LocalOptions = new HashMap<String, String>();
-            LocalOptions.put("key", accessKey);
-            local.start(LocalOptions);
-        }
-
-        driver = new AndroidDriver(new URL("http://"+config.get("server")+"/wd/hub"), options);
+        driver = new AndroidDriver(new URL("http://127.0.0.1:4723/wd/hub"), options);
```

Comment:
Use Remote Webdriver and document browserstack flow, so in case if javaagent argline is missing, user is still able to run on BrowserStack.